### PR TITLE
Unit test for jerk-limited trajectory generator

### PR DESCRIPTION
**Pull request by bresch**
First quick implementation of some tests for for the jerk-limited trajectory generator implemented in the VelocitySmoothing class.

### CODE REVIEW

File: src/lib/FlightTasks/tasks/Utility/VelocitySmoothingTest.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+#include "VelocitySmoothing.hpp"
+
+using namespace matrix;
+
+class VelocitySmoothingTest : public ::testing::Test
+{
+	public:
+		void SetUp() override
+		{
+			return;
```
Comment (jkflying) [inline src/lib/FlightTasks/tasks/Utility/VelocitySmoothingTest.cpp]: if you don't need any setup you can just delete this function

File: src/lib/FlightTasks/tasks/Utility/VelocitySmoothingTest.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	Vector3f x0(0.f, 0.f, 0.f);
+
+	setInitialConditions(a0, v0, x0);
+
+	// Generate the trajectories with zero setpoints
+	Vector3f velocity_setpoints(0.f, 0.f, 0.f);
+	float dt = 0.01f;
+
+	// Run a few times the algorithm
+	updateTrajectories(velocity_setpoints, dt);
```
Comment (jkflying) [inline src/lib/FlightTasks/tasks/Utility/VelocitySmoothingTest.cpp]: You're doing it in the loop, so I don't think you need another one here?

File: src/lib/FlightTasks/tasks/Utility/VelocitySmoothingTest.cpp [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+	// Check that time synchronization actually works
+	ASSERT_LE(fabsf(_trajectories[0].getTotalTime() - _trajectories[1].getTotalTime()), 0.0001);
+
+	// Compute the number of steps required to reach desired value
+	// because of known numerical issues, the actual trajectory takes a
+	// bit more time than the predicted one, this is why we have to add 14 steps
+	// to the theoretical value
+	float t123 = _trajectories[0].getTotalTime();
+	float nb_steps = ceilf(t123 / dt) + 14;
```
Comment (julianoes) [inline src/lib/FlightTasks/tasks/Utility/VelocitySmoothingTest.cpp]: `float`?

Comment (bresch) [reply src/lib/FlightTasks/tasks/Utility/VelocitySmoothingTest.cpp]: Oups

**Comment by bresch** [comment]
@jkflying I improved the PR, can you review again please?

**Pull request merged**